framework/PostProcessors/BasicStatistics.py

```diff
@@ -1124,12 +1128,18 @@ def sensitivityCalculation(self,featVars, targVars, featSamples, targSamples, in
           else:
             featMat = featSamples
           regCoeff = LinearRegression().fit(featMat, targSamples[:,p]).coef_
+          condNumber = np.linalg.cond(featMat)
+          if condNumber > 30.:
+            self.raiseAWarning("Condition Number: {:10.4f} > 30.0. Detected SEVERE multicollinearity problem. Sensitivity might be incorrect!".format(condNumber))
           if ind is not None:
             regCoeff = np.insert(regCoeff,ind,1.0)
           senMatrix[p,:] = regCoeff
     else:
       senMatrix = np.zeros((len(targVars), len(featVars)))
       for p, feat in enumerate(featVars):
+        condNumber = np.linalg.cond(featSamples[:,p])
+        if condNumber > 30.:
+          self.raiseAWarning("Condition Number: {:10.4f} > 30.0. Detected SEVERE multicollinearity problem. Sensitivity might be incorrect!".format(condNumber))
```

Comment:
There is no need to check it here. The sensitivity is calculated by looping over the feature space, basically, single feature is used here.

Reply:
removed